File: frontend/src/app/models/tour-log.ts
export interface TourLog {
  id: number;
  dateTime: string;          // ISO LocalDateTime, e.g. "2026-05-27T14:30:00"
  comment: string;
  difficulty: number;        // 1–5
  totalDistance: number;     // km
  totalTime: number;         // minutes (long on backend)
  rating: number;            // 1–5
}

export interface CreateTourLogRequest {
  dateTime: string;
  comment: string;
  difficulty: number;
  totalDistance: number;
  totalTime: number;
  rating: number;
}

File: backend/src/main/java/fhtw/swen2/model/TourLog.java
package fhtw.swen2.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "tour_logs")
@Getter
@Setter
@NoArgsConstructor
public class TourLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "tour_id", nullable = false)
    private Tour tour;

    @Column(nullable = false)
    private LocalDateTime dateTime;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @Column(nullable = false)
    private int difficulty;

    @Column(nullable = false)
    private double totalDistance;

    @Column(nullable = false)
    private long totalTime;

    @Column(nullable = false)
    private int rating;
}
